src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/VsLangProjectProperties.cs

```diff
@@ -0,0 +1,194 @@
+                return (prjOutputType)Convert.ToInt32(value.DisplayName);
+            }
+
+            set
+            {
+                var configurationGeneralBrowseObjectProperties = JoinableTaskFactory.Run(ProjectProperties.GetConfigurationGeneralBrowseObjectPropertiesAsync);
+                RunFuncTaskSynchronously(configurationGeneralBrowseObjectProperties.OutputType.SetValueAsync, (object)value);
+            }
+        }
+
+        public string AssemblyName
+        {
+            get
+            {
+                var configurationGeneralProperties = JoinableTaskFactory.Run(ProjectProperties.GetConfigurationGeneralPropertiesAsync);
+                return JoinableTaskFactory.Run(configurationGeneralProperties.AssemblyName.GetEvaluatedValueAtEndAsync);
+            }
+
+            set
+            {
+                var configurationGeneralProperties = JoinableTaskFactory.Run(ProjectProperties.GetConfigurationGeneralPropertiesAsync);
+                RunFuncTaskSynchronously(configurationGeneralProperties.AssemblyName.SetValueAsync, (object)value);
+            }
+        }
+
+        public string FullPath
+        {
+            get
+            {
+                var configurationGeneralProperties = JoinableTaskFactory.Run(ProjectProperties.GetConfigurationGeneralPropertiesAsync);
+                return JoinableTaskFactory.Run(configurationGeneralProperties.TargetPath.GetEvaluatedValueAtEndAsync);
+            }
+        }
+
+        public string ExtenderCATID => null;
+
+        public string AbsoluteProjectDirectory
+        {
+            get
+            {
+                var configurationGeneralBrowseObjectProperties = JoinableTaskFactory.Run(ProjectProperties.GetConfigurationGeneralBrowseObjectPropertiesAsync);
+                return JoinableTaskFactory.Run(configurationGeneralBrowseObjectProperties.FullPath.GetEvaluatedValueAtEndAsync);
+            }
+        }
+
+        public string __id => throw new NotImplementedException();
+
+        public object __project => throw new NotImplementedException();
+
+        public string StartupObject { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+
+        public string RootNamespace { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+
+        public string AssemblyOriginatorKeyFile { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+
+        public string AssemblyKeyContainerName { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+
+        public prjOriginatorKeyMode AssemblyOriginatorKeyMode { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+
+        public bool DelaySign { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
```

Comment:
DelaySign property is not from Common properties. It is backed up by `DynamicTypeBrowseObject` from [here](https://github.com/dotnet/roslyn-project-system/blob/master/src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/GeneralBrowseObject.xaml#L153). This will not be a problem.